CefSharp/DevTools/DevToolsClient.cs

```diff
@@ -78,27 +78,28 @@ public void SetDevToolsObserverRegistration(IRegistration devToolsRegistration)
         }
 
         /// <inheritdoc/>
-        public IRegistration RegisterEventHandler<T>(string eventName, EventHandler<T> eventHandler) where T : DevToolsDomainEventArgsBase
+        public void AddEventHandler<T>(string eventName, EventHandler<T> eventHandler) where T : EventArgs
         {
-            EventHandler<Stream> handler = (sender, stream) =>
+            if (eventHandlers.TryGetValue(eventName, out IEventProxy eventProxy))
             {
-                stream.Position = 0;
-                if (typeof(T) == typeof(DevToolsEventArgs) || typeof(T) == typeof(DevToolsDomainEventArgsBase))
-                {
-                    var paramsAsJsonString = StreamToString(stream, leaveOpen: true);
-                    var args = new DevToolsEventArgs(eventName, paramsAsJsonString);
-
-                    eventHandler(sender, (T)(object)args);
-                }
-                else
-                {
-                    eventHandler(sender, DeserializeJson<T>(stream));
-                }
-            };
-
-            eventHandlers.AddOrUpdate(eventName, _ => handler, (_, existingHandler) => existingHandler += handler);
+                var p = ((EventProxy<T>)eventProxy);
+                p.AddHandler(eventHandler);
+            }
+            else
+            {
+                eventProxy = new EventProxy<T>(eventHandler, DeserializeJsonEvent<T>);
+                eventHandlers.TryAdd(eventName, eventProxy);
+            }
+        }
```

Comment:
This code could be reduced to this, which would be save for cuncurrent usage.

```c#
        public void AddEventHandler<T>(string eventName, EventHandler<T> eventHandler) where T : EventArgs
        {
            var eventProxy = eventHandlers.GetOrAdd(eventName, _ => new EventProxy<T>(DeserializeJsonEvent<T>));

            var p = (EventProxy<T>)eventProxy;
            p.AddHandler(eventHandler);
        }
```

Reply:
Thanks 👍 Updated